Datastore/src/Operation.php

```diff
@@ -614,12 +750,40 @@ public function commit(array $mutations, array $options = [])
             'databaseId' => $this->databaseId,
         ];
 
-        $res = $this->connection->commit($options + [
-            'mode' => ($options['transaction']) ? 'TRANSACTIONAL' : 'NON_TRANSACTIONAL',
+        foreach ($mutations as &$mutation) {
+            $mutationType = array_keys($mutation)[0];
+            $data = $mutation[$mutationType];
+            if (isset($data['properties'])) {
+                foreach ($data['properties'] as &$property) {
+                    list($type, $val) = $this->toGrpcValue($property);
+
+                    $property[$type] = $val;
+                }
+            }
+
+            $mutation[$mutationType] = $data;
+
+            $mutation = $this->serializer->decodeMessage(new Mutation(), $mutation);
+        }
+
+        $options += [
+            'mode' => ($options['transaction']) ? Mode::TRANSACTIONAL : Mode::NON_TRANSACTIONAL,
             'mutations' => $mutations,
             'projectId' => $this->projectId,
-        ]);
+        ];
 
+        if (is_null($options['transaction'])) {
+            // Remove 'transaction' if set to `null` to avoid serialization error
+            unset($options['transaction']);
+        }
+        list($data, $optionalArgs) = $this->splitOptionalArgs($options, ['allowOverwrite', 'baseVersion']);
```

Comment:
Same question as one above, are these supposed to go as an optional arg in the call options?

Reply:
No, we just need to remove them from the data send to protobuf descriptors constructor. I've implemented in a better way, please see https://github.com/yash30201/google-cloud-php/pull/4.